File: fixtures/javascript/put_basic_auth_json_data.js
fetch('http://localhost:5984/test/_security', {
    method: 'PUT',
    headers: {
        'Content-Type': 'application/x-www-form-urlencoded',
        'Authorization': 'Basic ' + btoa('admin:123')
    },
    body: JSON.stringify({"admins":{"names":[], "roles":[]}, "readers":{"names":["joe"],"roles":[]}})
});

File: fixtures/r/put_basic_auth_json_data.r
require(httr)

headers = c(
  `Content-Type` = 'application/x-www-form-urlencoded'
)

data = '{"admins":{"names":[], "roles":[]}, "readers":{"names":["joe"],"roles":[]}}'

res <- httr::PUT(url = 'http://localhost:5984/test/_security', httr::add_headers(.headers=headers), body = data, httr::authenticate('admin', '123'))
